portal-impl/src/com/liferay/portlet/blogs/service/impl/BlogsEntryLocalServiceImpl.java

```diff
@@ -684,10 +684,26 @@ public int getCompanyEntriesCount(
 
 		BlogsEntry entry = blogsEntryPersistence.findByPrimaryKey(entryId);
 
-		return blogsEntryPersistence.findByG_S_PrevAndNext(
-			entry.getEntryId(), entry.getGroupId(),
-			WorkflowConstants.STATUS_APPROVED,
-			new EntryDisplayDateComparator(true));
+		long groupId = entry.getGroupId();
+
+		Date displayDate = entry.getDisplayDate();
+
+		BlogsEntry[] blogsEntries =
+			blogsEntryPersistence.findByG_S_D_PrevAndNext(
+				entryId, groupId, WorkflowConstants.STATUS_APPROVED,
+				displayDate, new EntryDisplayDateComparator(true));
+
+		if (Validator.isNull(blogsEntries[0])) {
```

Comment:
Since blogEntries[0] is not a String, we should better do `blogEntries[0] == null`